### PR TITLE
Open target meganav dropdown based on url hash on load

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Adds a function 'handleUrlHash', that on load, will check if the url has a hash (ex. #products) and open the corresponding dropdown

## QA

- Go to  and see that nothing opens
- Go to https://ubuntu-com-13546.demos.haus/#products and see that the 'products' dropdown opens
- Go to https://ubuntu-com-13546.demos.haus/download/desktop#use-case and see that the 'use-case' dropdown opens
- Go to https://ubuntu-com-13546.demos.haus/#made-up-hash and see that nothing opens
- check normal functionality is retained

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-8727

### CODE REVIEW

**Comment by juanruitina** [comment]
I see that it doesn't clash with other anchor links, such as https://ubuntu-com-13546.demos.haus/download/server#downloads

However, weird behaviour on mobile: only the overlay opens, not the menu.

**Comment by petesfrench** [comment]
@juanruitina can you take another look please

**Comment by juanruitina** [comment]
UX+1!

File: static/js/src/navigation.js [hunk 1]
```diff
@@ -139,6 +144,22 @@ function updateUrlHash(id, open) {
...
 
+function handleUrlHash() {
+  const targetId = window.location.hash;
+  const targetDropdown = targetId ? navigation.querySelector(targetId) : null;
+  if (targetDropdown) {
+    const currViewportWidth = window.innerWidth;
+    const isMobile = currViewportWidth < MOBILE_VIEW_BREAKPOINT
+    if (isMobile) {
+      const menuToggle = navigation.querySelector(".js-menu-button");
+      console.log("menuToggle", menuToggle);
```
Comment (akbarkz) [inline static/js/src/navigation.js:155]: remove?

**Comment by akbarkz** [comment]
@petesfrench When opening  https://ubuntu-com-13546.demos.haus/download/desktop#use-case on a mobile device (I used Android phone) it opens the Use Cases menu, but when you scroll down it doesn't show Telco and Robotics is half cut (like from the bug that you fixed before)

**Review by akbarkz — APPROVED** [review]

**Pull request merged**